### PR TITLE
[Grafana][Observability] Embed Grafana dashboard panels into Ray dashboard

### LINKED ISSUE #1020: [Docs] Improve metrics/grafana docs to include info on how to utilize the new Ray dashboard features.
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

Others

### What happened + What you expected to happen

The current grafana / prometheus docs works well for folks using grafana UI directly. However, in ray 2.2 and above there are metrics features embedded into the Ray Dashboard UI and those features should work with kube-ray's grafana deployment.

### Reproduction script

None. Just launch a cluster and check the ray dashboar.

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

* Replace `dashboard_default.json` with `default_grafana_dashboard.json` (generated by Ray 2.5.0)

* `install/prometheus/install.sh` & `overrides.yaml`: Pin the `kube-prometheus-stack` chart to version v48.2.1 and create the chart using `overrides.yaml` to configure Grafana. This will enable the embedding of Grafana dashboard panels into the Ray dashboard.

* `ray-cluster.embed-grafana.yaml`: An example YAML file to demonstrate embedding

## Related issue number

Closes #1020 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: config/grafana/default_grafana_dashboard.json [hunk 1]
```diff
@@ -0,0 +1,2820 @@
...
+                "align": false,
+                "alignLevel": null
+            }
+        }
+    ],
+    "refresh": false,
+    "schemaVersion": 27,
+    "style": "dark",
+    "tags": [
+        "rayVersion:2.5.0"
```
Comment (dcela) [inline config/grafana/default_grafana_dashboard.json:2738]: should this be 2.6.1?

It has some differences from the default dashboard generated automatically from 2.6.1

Comment (kevin85421) [reply config/grafana/default_grafana_dashboard.json:2738]: Yes, that's why I mentioned that manually importing the dashboard, as stated in the "TODO: Note that importing the dashboard manually is not ideal", isn't ideal. We should find a way to import the Grafana dashboard automatically in the future. I added some sentences to describe it https://github.com/ray-project/kuberay/pull/1278/commits/b94b5d63113e3a5f1ee8049183465f4b7aca3b12. Does the change make sense to you?

**Comment by kevin85421** [comment]
cc @alanwguo would you mind reviewing this PR? Thanks!

File: docs/guidance/prometheus-grafana.md [hunk 2]
```diff
@@ -66,6 +80,22 @@ kubectl get service
...
+      value: http://127.0.0.1:3000
+    - name: RAY_GRAFANA_HOST
+      value: http://prometheus-grafana.prometheus-system.svc:80
+    - name: RAY_PROMETHEUS_HOST
+      value: http://prometheus-kube-prometheus-prometheus.prometheus-system.svc:9090
+  ```
+  * Note that we do not deploy Grafana in the head Pod, so we need to set both `RAY_GRAFANA_IFRAME_HOST` and `RAY_GRAFANA_HOST`. 
+    `RAY_GRAFANA_HOST` is used by the head Pod to send health-check requests to Grafana in the backend.
+    `RAY_GRAFANA_IFRAME_HOST` is used by your browser to fetch the Grafana panels from the Grafana server rather than from the head Pod.
+    Because we forward the port of Grafana to `127.0.0.1:3000` in this example, we set `RAY_GRAFANA_IFRAME_HOST` to `http://127.0.0.1:3000`.
```
Comment (alanwguo) [inline docs/guidance/prometheus-grafana.md:96]: should we include instructions that if user does not want to port-forward, they should set up an ingress?

Comment (kevin85421) [reply docs/guidance/prometheus-grafana.md:96]: We have instructions for setting up the ingress for the Ray dashboard. Setting up the ingress for Grafana seems to be beyond the scope of KubeRay. How about adding https://grafana.com/tutorials/run-grafana-behind-a-proxy/ to the doc?

Comment (kevin85421) [reply docs/guidance/prometheus-grafana.md:96]: Add the link in https://github.com/ray-project/kuberay/pull/1278/commits/7362c589d64fd9a15ad3a135538c96ba60351c00. Does it make sense to you? Thx!

**Review by alanwguo — APPROVED** [review]

**Pull request merged**